### PR TITLE
Add release notes to Zulip bot announcements

### PR DESCRIPTION
**Pull request by lgan-czi**
Background:

When a napari-plugin has a new release, a message should appear in the napari-hub-bot zulip chat alerting community members of an update.  However, the bot doesn't post any information about that new plugin, and community members have to manually navigate to the repo in order to learn about the update. This process is inefficient and promotes a bad user experience and explicitly including release notes could motivate developers to maintain them, increasing the health of the plugin community.

Changes:

I modified the `update_cache()` function in model.py to pass metadata information to the zulip bot. This information is then used to help the zulip bot find release notes from the github API and include them in it's messages. If it doesn't find anything, it will not include anything as a release note.

Testing:

Added some unit tests to test a new methods I added to help get messages from the github API and test the logic of the message generation. I also tested getting information from the github API on a jupyter notebook and posted the results here.

Screen Shot 2022-07-01 at 3 12 02 PM

I then created a jupyter notebook to directly send myself zulip bot messages and was able to post messages to myself as seen in the screenshot below. Those selections of plugins cover the spectrum of the types of plugins I expect the bot to deal with when sending messages. i tested messages for plugins with v and without v in their version number, plugins with and without release notes, and plugins with and without code repositories.
Screen Shot 2022-07-13 at 11 55 47 AM

### CODE REVIEW

File: backend/api/zulip.py [hunk 1]
```diff
@@ -67,3 +62,81 @@ def send_zulip_message(username: str, key: str, topic: str, message: str):
...
+    :param release_notes: release notes found from the github api
+    :param link_to_release: a link to the github release page, if it exists
+    """
+    if package not in existing_packages:
+        if not release_notes:
+            link_to_release = ''
+        else:
+            link_to_release = '\nAlso check out its release notes for version ' + link_to_release + '!\n\n'
+        # this string needs reformatting
+        message = f'A new plugin has been published on the napari hub! ' \
```
Comment (potating-potato) [inline backend/api/zulip.py:132]: curious why are we using the formatting string together with concatenation? why not just the format string with f'{link_to_release}{release_notes}' instead?

**Comment by potating-potato** [comment]
I think this is a good PR for Richa to help review, can you please tag her?

**Comment by richaagarwal** [comment]
Thanks for tagging me! Is there a corresponding github issue for this PR?

**Comment by lgan-czi** [comment]
Yep! It's related to [this](https://github.com/chanzuckerberg/napari-hub/discussions/492)

File: backend/api/zulip.py (hunk 1 above)
Comment (lgan-czi) [reply backend/api/zulip.py:132]: Yeah formatting is way better, switching now

**Review by potating-potato — APPROVED** [review]
at a high-level glance, and based on the screenshot of the behavior this seems ok. I will defer the detailed review to Richa, I think this is a good opportunity to dive into this small well-scoped function area (sending announcements to zulip)

**Comment by richaagarwal** [comment]
Thanks @potating-potato and @lgan-czi  - I'll make time to review this early next week, if not sooner.

**Comment by richaagarwal** [comment]
@lgan-czi @potating-potato I took a cursory look at this - there are a lot of changes here (some of which I'd like to give more thoughtful feedback to) so I don't feel comfortable approving and moving forward without a more detailed review. 

Given that this work wasn't discussed in sprint planning and would require some time from the hub team, I'd like to defer a review until more hub members are settled back in from PTO and we can prioritize upcoming work. I'm happy to chat more about any of this but want to be mindful of the priorities we already committed to as a team.

File: backend/api/zulip.py [hunk 2]
```diff
@@ -1,41 +1,38 @@
...
-        elif existing_packages[package] != version:
-            message = f'A new version of [{package}](https://napari-hub.org/plugins/{package}) is available on the ' \
-                      f'napari hub! Check out [{version}](https://napari-hub.org/plugins/{package})!'
-            if username and key:
-                send_zulip_message(username, key, package, message)
-            else:
-                print(message)
+        release_notes, link_to_release = generate_release_notes_and_link_to_release(package, version, packages_metadata)
+        message = create_message(package, version, existing_packages, release_notes, link_to_release)
+        if username and key and message:
```
[the remark this answers is not in the document]
Comment (lgan-czi) [reply backend/api/zulip.py:32]: Oh wait, I should've been more clear about this part of the code, I do send a message (A new plugin has been published on the napari hub! Check out [{package}](https://napari-hub.org/plugins/{package})!)) if there's no release notes. What I don't send a message for is whenever there hasn't been a change in version. I'll edit the code to make that a bit more clear! I also just realized I didn't add a unit test testing what happens if the version doesn't change, so I'll also work on that for a bit now.

**Comment by lgan-czi** [comment]
Sounds good!

File: backend/api/zulip.py [hunk 3]
```diff
@@ -45,10 +40,96 @@ def notify_new_packages(existing_packages: Dict[str, str], new_packages: Dict[st
...
+    """
+    Parses through the metadata of a plugin to find it's release notes if they exist
+    If they don't exist, return a blank string and a link to napari-hub
+    Returns the release notes and a link for the zulip bot to add to its message
+
+    :param existing_packages: existing packages in cache
+    :param new_packages: new packages found
+    :param packages_metadata: metadata for the packages, contains information about github links
+    """
+    if "code_repository" in packages_metadata[package] and packages_metadata[package]["code_repository"]:
```
Comment (lgan-czi) [inline backend/api/zulip.py:87]: Use .get

**Comment by lgan-czi** [comment]
Thanks for the chat @richaagarwal! I'll be implementing some of the changes regarding documentation and improving readability!

**Review by richaagarwal — APPROVED** [review]
Thanks so much for your work on this and addressing all my feedback, @lgan-czi!  

Looks great overall. I left one comment on an unused variable in `test_utils.py` - I think there are several other unused variables in there as well. Are there some missing updates to `test_github.py`?

File: backend/utils/test_utils.py [hunk 4]
```diff
@@ -145,6 +145,82 @@ def raise_for_status(self):
 version: "0.0.0"
 """
 
+# constants for testing parsing author info from citation
+citation_string_no_auth_name = """# YAML 1.2
```
Comment (richaagarwal) [inline backend/utils/test_utils.py:149]: It looks like this variable isn't used anywhere - should it be removed?

**Pull request merged**